renku/core/util/dispatcher.py

```diff
@@ -52,6 +53,16 @@ def get_database_helper(database_dispatcher: IDatabaseDispatcher):
     return get_database_helper()
 
 
+def get_dataset_gateway() -> "DatasetGateway":
```

Comment:
I'm honestly not sure I'm a big fan of these types of methods getting added everywhere. It might make the method body more readable, but it does lose the big plus where in testing you can just pass in a mocked instance of an injected class directly to test the method in isolation. Whereas this makes it so we must always set up fake injection when testing single methods.

I quite like being able to do https://github.com/SwissDataScienceCenter/renku-python/blob/develop/tests/core/commands/test_graph.py#L318-L323

Especially if the mocked instances have to be changed between calls, updating injection/replacing existing entries is a nightmare compared to just passing another instance.

Reply:
TBH, this is the first test that I see which doesn't use injection. Testing like this is possible only if the method doesn't call other method that use injection. Otherwise, we still need to set up a fake injection.

I've changed its usage in my code since it doesn't call other injected method.